File: Python/5 - Função/Questões/q.3.py
# Implemente uma função recursiva chamada fatorial que calcule o fatorial de um número n. 
# Lembre-se que o fatorial de 0 é 1, e para outros números, n! = n * (n-1)!.

def fatorial(numero):
    if numero == 0:
        return 1
    if numero == 1:
        return 1
    else:
        return numero * fatorial(numero - 1)

numero = int(input("Digite o numero que deseja saber o fatorial: "))
print(f"\nFatorial de {numero} = {fatorial(numero)}")

File: C/3 - Repetição/Questões do livro resolvidas/q.3.c
#include <stdio.h>

int main(){
    int i,vl,n,f;
    printf("Insira quantos valores devem ser lidos:");
    scanf("%d",&vl);
    for (i = 0; i < vl; i++){
        printf("Digite um numero que deseja saber o fatorial:");
        scanf("%d",&n);
        f=1;
        for (int j = 1; j <= n; j++){
            f*=j;
        }
        printf("O Fatorial do numero %d eh %d\n",n,f);
    }
    return 0;
}